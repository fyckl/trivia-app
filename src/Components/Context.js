import React, { useContext, useState, useEffect } from 'react'

const Context = React.createContext()

function ContextProvider(props) {

    const [startQuiz, setStartQuiz] = useState(false)
    const [arrQuestions, setArrQuestions] = useState([])
    const [playAgain, setPlayAgain] = useState(false)
    const [correctAnswers, setCorrectAnswers] = useState(0)
    const [score, setScore] = useState(0)
    const [shuffleAnswers, setShuffleAnswers] = useState(false)

    function quizStart(){
      setShuffleAnswers(true)
      setStartQuiz(true)
    }

    function checkAnswers(){
      setShuffleAnswers(false)
      setPlayAgain(true)
    }

    useEffect(() => {
      setScore(prevState => prevState + correctAnswers)
    }, [correctAnswers])

    function reset(){
      setArrQuestions([])
      getQuestions()
      setPlayAgain(false)
      setCorrectAnswers(0)
      setShuffleAnswers(true)
    }

    function getQuestions(){
      fetch('https://opentdb.com/api.php?amount=5&difficulty=easy&type=multiple&encode=base64')
      .then(res=> res.json())
      .then(data=> setArrQuestions(data.results))
    }

    
    // console.log(arrQuestions)
    useEffect(() => {
      getQuestions()
    }, [])

    return(
        <Context.Provider 
            value={{
                quizStart, 
                startQuiz, 
                arrQuestions,
                setCorrectAnswers,
                checkAnswers,
                playAgain,
                correctAnswers,
                reset,
                score,
                shuffleAnswers
        }}>
            {props.children}
        </Context.Provider>
    )
}

export {ContextProvider, Context}