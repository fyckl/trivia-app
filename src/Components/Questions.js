import React, { useContext } from 'react'
import Question from './Question'
import useMappedQuestions from '../Hooks/useMappedQuestions'
import { Context, useAppContext } from "./Context"

export default function Questions(props) {
  const {
    arrQuestions, 
    checkAnswers, 
    isPlayAgain, 
    correctAnswers, 
    reset, 
    score
  } = useAppContext()

  // function questions(arr){ arr.map(item => (
  //   <Question 
  //     key={item.question}
  //     question={item.question} 
  //     correctAnswer={item.correct_answer} 
  //     incorrectAnswers={item.incorrect_answers}
  //   />
  // ))
  // }
  
  return (
    <div className='text-center questions-container' data-testid="container">
      {useMappedQuestions(arrQuestions)}

      {
        isPlayAgain ? 
        <div className='flex' id="post-game-details-container">
          <h4 className='inter'>You got {correctAnswers}/5 questions right</h4>
          <button className='again-button' onClick={reset}>Play again</button>
        </div> : 
        <button className='check-button' onClick={checkAnswers} id="check-answers">Check answers</button>
      }

      <h4 className='inter score'>Score: {score}</h4>
    </div>
  )
}
