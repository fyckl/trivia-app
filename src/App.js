import React, { useState, useEffect, useContext } from "react";
import Start from "./Components/Start";
import Questions from "./Components/Questions";
import Answers from "./Components/Answer";
import { Context } from "./Components/Context"
function App() {
  const {quizStart, startQuiz} = useContext(Context)

  return (
    <div className="main-container">
      {startQuiz ? <Questions /> : <Start start={quizStart}/>}
    </div>
  );

}

export default App;
