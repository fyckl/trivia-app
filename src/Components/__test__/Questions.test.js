import React from "react"
import Questions from "../Questions"
import * as Context  from "../Context"
import App from "../../App"
import { fireEvent, cleanup, screen, render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import { Enzyme, shallow, configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
// let getByTestId

// beforeEach(() => {
//     const component = render(
//         <ContextProvider>
//             <App>
//                 <Questions />
//             </App>
//         </ContextProvider>
//     )
//     getByTestId = component.getByTestId

// })


// test("Container renders with correct classNames", () => {
//     const containerEl = getByTestId("container")
//     console.log(containerEl.children)
//     expect(containerEl.className).toBe("text-center questions-container")
// })

// test("enzyme dive", () => {
//     const TestComponent = () => (
//         <NameContext.Provider value="Provided Value">
//             <MyComponent />
//         </NameContext.Provider>
//     );
//     const element = shallow(<TestComponent />);
//     expect(element.find(MyComponent).dive().text()).toBe("Provided Value");

test("renders correctly to match snapshot", () => {
    const contextValues = {
        arrQuestions: [], 
        checkAnswers: [], 
        isPlayAgain: true, 
        correctAnswers: [],
        reset: jest.fn(), 
        score: 0
    }

    jest
      .spyOn(Context, 'useAppContext')
      .mockImplementation(() => contextValues);

    const wrapper = shallow(<Questions/>)
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find(Questions)).toBeTruthy();

   
})


test("check answers button conditonally renders", () => {
    const contextValues = {
        arrQuestions: [], 
        checkAnswers: [], 
        isPlayAgain: false, 
        correctAnswers: [],
        reset: jest.fn(), 
        score: 0
    }
    jest
      .spyOn(Context, 'useAppContext')
      .mockImplementation(() => contextValues);

    const wrapper = shallow(<Questions/>)
    expect(wrapper.find("#check-answers")).toHaveLength(1)
})