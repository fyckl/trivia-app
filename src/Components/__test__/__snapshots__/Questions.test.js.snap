// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly to match snapshot 1`] = `"<div class=\\"text-center questions-container\\" data-testid=\\"container\\"><div class=\\"flex\\" id=\\"post-game-details-container\\"><h4 class=\\"inter\\">You got /5 questions right</h4><button class=\\"again-button\\">Play again</button></div><h4 class=\\"inter score\\">Score: 0</h4></div>"`;
